ENIGMAsystem/SHELL/Graphics_Systems/General/GSmatrix.cpp

```diff
@@ -247,17 +247,17 @@ void d3d_set_projection_ortho_lookat(gs_scalar x, gs_scalar y, gs_scalar width,
                                      gs_scalar xto, gs_scalar yto, gs_scalar zto,
                                      gs_scalar xup, gs_scalar yup, gs_scalar zup) {
   enigma::view = glm::lookAt(glm::vec3(xfrom, yfrom, zfrom), glm::vec3(xto, yto, zto), glm::vec3(xup, yup, zup));
+  enigma::view = glm::rotate(glm::mat4(1.0f), (float)gs_angle_to_radians(angle), glm::vec3(0.0f, 0.0f, 1.0f)) * enigma::view;
   enigma::projection = glm::ortho(x, x + width, y + height, y, -32000.0f, 32000.0f);
-  enigma::projection = glm::rotate(glm::mat4(1.0f), (float)gs_angle_to_radians(angle), glm::vec3(0.0f, 0.0f, 1.0f)) * enigma::projection;
   enigma::graphics_set_matrix(matrix_view);
   enigma::graphics_set_matrix(matrix_projection);
 }
 
 void d3d_set_projection_perspective(gs_scalar x, gs_scalar y, gs_scalar width, gs_scalar height, gs_scalar angle)
 {
-  enigma::view = glm::ortho(x, x + width, y + height, y, 1.0f, 32000.0f);
+  enigma::view = glm::translate(glm::mat4(1.0f), glm::vec3(-(x + width/2), -(y + height/2), width));
```

Comment:
Please average the width and height for z, or choose the max, unless this is somehow destructive to compatibility.

Reply:
Ok: 4e3f55b68a6868b743db5fed45fd9390c116b61b

Fine by me, works with Project Mario, which is the only game I've found to date that truly uses this function. All of the other games never actually do due to the poor documentation and highly misunderstood behavior of the perspective functions in GM's API.